src/views/pages/profiles/LoginPageTs.ts

```diff
@@ -118,21 +119,17 @@ export default class LoginPageTs extends Vue {
         // filter out invalid profiles
         this.profiles = this.profileService.getProfiles().filter((p) => p.accounts.length > 0);
 
-        const reducer = (accumulator: { networkType: NetworkType; profiles: ProfileModel[] }[], currentValue: ProfileModel) => {
-            const currentAccumulator = accumulator.find((a) => a.networkType == currentValue.networkType);
-            if (currentAccumulator) {
-                currentAccumulator.profiles.push(currentValue);
-                return accumulator;
-            } else {
-                return [...accumulator, { networkType: currentValue.networkType, profiles: [currentValue] }];
-            }
-        };
-
-        this.profilesClassifiedByNetworkType = this.profiles.reduce(reducer, []);
         if (!this.profiles.length) {
             return;
         }
-        // accounts available, iterate to first profiles
+
+        const profilesGroupedByNetworkType = _.groupBy(this.profiles, (p) => p.networkType);
+        this.profilesClassifiedByNetworkType = Object.values(profilesGroupedByNetworkType).map((profiles) => ({
+            networkType: profiles[0].networkType,
```

Comment:
only the top one?

Reply:
Since `profilesGroupedByNetworkType` is grouped by network type, all the profiles have the same network type... 😀 
So yes, the network type can be retrieved from the first profile.